main.go

```diff
@@ -263,30 +272,110 @@ func (svr *bridge) handleCall(w http.ResponseWriter, r *http.Request) {
 			}
 
 			if val, ok := alert.Annotations[*svr.titleAnnotation]; ok {
-				title += val
+				titleTemplate, err := template.New("title").Parse(val)
+				if err != nil {
+					proceed = false
+					errMsg := fmt.Sprintf("Error in Template: %s", err)
+					text = []string{errMsg}
+					respCode = http.StatusBadRequest
+					if *svr.debug {
+						log.Println(errMsg)
+					}
+					if *svr.dispatchErrors {
+						proceed = true
+						title = "Alertmanager-Gotify-Bridge Error"
+						message = fmt.Sprintf("Error: %s\n\nAlso check Alertmanager, maybe an alert was raised!\n\nIcomming request:\n%s", errMsg, b)
+					}
+				} else {
+					var templatedTitle bytes.Buffer
+					err = titleTemplate.ExecuteTemplate(&templatedTitle, "title", alert)
+					if err != nil {
+						proceed = false
+						errMsg := fmt.Sprintf("Error in Template: %s", err)
+						text = []string{errMsg}
+						respCode = http.StatusBadRequest
+						if *svr.debug {
+							log.Println(errMsg)
+						}
+						if *svr.dispatchErrors {
+							proceed = true
+							title = "Alertmanager-Gotify-Bridge Error"
+							message = fmt.Sprintf("Error: %s\n\nAlso check Alertmanager, maybe an alert was raised!\n\nIcomming request:\n%s", errMsg, b)
+						}
+					} else {
+						title += templatedTitle.String()
+					}
+				}
+
 				if *svr.debug {
-					log.Printf("    title: %s\n", title)
+					log.Printf("title: %s\n", title)
 				}
 			} else {
 				proceed = false
-				text = []string{fmt.Sprintf("Missing annotation: %s", *svr.titleAnnotation)}
+				errMsg := fmt.Sprintf("Missing annotation: %s", *svr.titleAnnotation)
+				text = []string{errMsg}
 				respCode = http.StatusBadRequest
 				if *svr.debug {
-					log.Printf("    title annotation (%s) missing\n", *svr.titleAnnotation)
+					log.Println(errMsg)
+				}
+				if *svr.dispatchErrors {
+					proceed = true
+					title = "Alertmanager-Gotify-Bridge Error"
+					message = fmt.Sprintf("Error: %s\n\nAlso check Alertmanager, maybe an alert was raised!\n\nIcomming request:\n%s", errMsg, b)
 				}
 			}
 
 			if val, ok := alert.Annotations[*svr.messageAnnotation]; ok {
-				message = val
+				messageTemplate, err := template.New("message").Parse(val)
```

Comment:
Since these two blocks of code are largely the same and have a number of branches in the logic, I think it would make sense to drop it into a single function. Maybe something like

```
func renderTemplate(name string, alert Alert) (string, error) {
...
}
```

Which could replace this section and reduce the number of times to check for error and take alternative action if `*svr.dispatchErrors` and modify title/message. What do you think?

Reply:
Good point. With a function this is better in all regards.